include/libtorrent/aux_/session_impl.hpp

```diff
@@ -1018,6 +1020,15 @@ namespace aux {
 #if TORRENT_USE_I2P
 			i2p_connection m_i2p_conn;
 			std::optional<socket_type> m_i2p_listen_socket;
+			
+			// Timer for I2P reconnection attempts
+			deadline_timer m_i2p_reconnect_timer;
+			
+			// Exponential backoff state for I2P reconnection
+			int m_i2p_reconnect_delay = 1;  // seconds
```

Comment:
```suggestion
			seconds m_i2p_reconnect_delay = seconds(1);
```

(and make sure `libtorrent/time.hpp` is included)